Add extra Kompose documentation

cdrage opened the pull request:
Create additional files for Kompose documentation including
architecture, development, a more extensive index and a user guide.

This also bumps Kompose up to the top on tools page to be symmetrical to
the what's listed on the side navbar.

I have assigned myself to each doc in order to get updates / any
questions regarding the updating of documentation.

<!-- Reviewable:start -->

---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1607)
<!-- Reviewable:end -->

cdrage commented:
ping @ngtuna @tkral @janetkuo

janetkuo commented on docs/tools/kompose/user-guide.md:
````diff
@@ -0,0 +1,307 @@
+For example:
+
+```yaml
+--- 
````
remove `---`

janetkuo commented on docs/tools/kompose/user-guide.md:
```diff
@@ -0,0 +1,307 @@
+| kompose.service.type | nodeport / clusterip / loadbalancer |
+
+
+For example:
```
Somehow this part seems unfinished. Maybe change "for example" to "Here's an example docker-compose.yml file that ..."

janetkuo commented on docs/tools/kompose/user-guide.md:
````diff
@@ -0,0 +1,307 @@
+```
+Note:
+- You must have a running Kubernetes cluster with a pre-configured kubectl context.
+- Only deployments and services are generated and deployed to Kubernetes. If you need different kind of resources, use the 'kompose convert' and 'kubectl create -f' commands instead.
````
These bullet points aren't rendered well on the website.

janetkuo replied:
Change these to `*` to see if it works? Or add a newline after "Note:"

janetkuo commented on docs/tools/kompose/user-guide.md:
````diff
@@ -0,0 +1,307 @@
+```
+
+Note:
+- You must have a running OpenShift cluster with a pre-configured `oc` context (`oc login`)
````
here too

janetkuo commented on docs/tools/kompose/development.md:
```diff
@@ -0,0 +1,145 @@
+---
```
This doc is for kompose developers. We only put this kind of docs in github repo.

janetkuo commented on docs/tools/kompose/architecture.md:
```diff
@@ -0,0 +1,84 @@
+---
```
This doc is for kompose developers. We only put this kind of docs in github repo.

janetkuo commented:
@kubernetes/docs @devin-donnelly @pwittrock thoughts on adding kompose docs under "tools"?

cdrage commented:
Thanks @janetkuo i'll update this and remove the relevant "architecture" and "development" examples, you're right. that should be directed to the github repo instead.

In terms of other pages, I think that the "overview" page, "user guide" page as well as an example / getting started page would be enough of a decent introduction to Kompose.

cdrage commented:
@janetkuo I've updated the PR with the changes you've suggest.

cdrage commented:
ping @kubernetes/docs

cdrage commented:
pingarino @kubernetes/docs @devin-donnelly @pwittrock :)

devin-donnelly commented:
I'm not a fan of just updating the Tools TOC with Kompose alone--the rest of the tools need to be on equal footing.

@janetkuo, would you mind building out the TOC to include the built-in tools as well as Helm after this is merged?

cdrage commented:
@janetkuo any changes you wish for me to make? was unfamiliar with the Tools TOC so I added it in manually (is it auto generated?)

devin-donnelly commented:
@cdrage, I'm okay with adding Kompose to the tools ToC, but we need to add the native Kubernetes tools above it and Helm below it. If we don't do that, it looks like Kompose is the only supported tool, which I don't want.

I can take that on myself, I suppose, since nobody else is jumping to do it.

cdrage commented:
@devin-donnelly Okay :) Are you able to open an issue and assign yourself to it? Unfortunately I can't do anything assign wise / label wise. 

We can get some Helm PR's up / other third-party tools so Kompose docs isn't lonely up there on the Tools page.

devin-donnelly commented:
Hi @cdrage , sorry this took so long (got bogged down with 1.5 launch docs), but I've managed to update the Tools TOC and restructure the page slightly in PRs #2075 and #2076 .

This, by necessity, introduced some merge conflicts into tools.yml, but you should be able to resolve those and build on the structure there. Feel free to do so and I'll be happy to merge this PR for you.

cdrage commented:
@devin-donnelly Not a problem / thanks for everything! :)

I've gone ahead and updated the PR (specifically, removed the conflicting files with `tools.yml`).

devin-donnelly commented:
@cdrage, feel free to update _tools.yml and the Tools index page to point to your new documentation; I didn't do that because I didn't want to publish the links that wouldn't exist until this PR got merged.

cdrage commented:
@devin-donnelly Ah crap, so I *was* suppose to keep those changes in. I'll update it.

cdrage commented:
@devin-donnelly I've updated the index / tools. Let me know if this is correctly done.

cdrage commented:
@devin-donnelly so I ended up updating it but I'm experiencing a weird issue.

Upon clicking on "Kompose" on the list to the side of the tools index. It takes me to the proper Kompose index file, however, the submenu on the left thinks it's on the "support" tab.

janetkuo commented:
> Upon clicking on "Kompose" on the list to the side of the tools index. It takes me to the proper Kompose index file, however, the submenu on the left thinks it's on the "support" tab.

You'll need to update `_data/tools.yml` and `docs/tools/index.md` to make kompose docs be linked correctly

cdrage commented:
@janetkuo Yup, I already did. But I'm not too sure what I'm doing wrong. Can you check the code that's commited in this PR?

devin-donnelly commented on docs/tools/kompose/index.md:
```diff
@@ -0,0 +1,41 @@
+---
+assignees:
+- cdrage
+
```
Please add an explicit title line here. Something like:

title: Kompose

devin-donnelly commented on docs/tools/kompose/user-guide.md:
```diff
@@ -0,0 +1,308 @@
+---
+assignees:
+- cdrage
+
```
Please add an explicit title.

title: Kompose User Guide

janetkuo commented on _data/tools.yml:
```diff
@@ -17,5 +17,4 @@ toc:
   - title: Kompose
-    path: https://github.com/kubernetes-incubator/kompose
-  
+    path: /docs/tools/kompose/
```
I think I know what's causing the problem here. Due to some recent GitHub Pages changes, we no longer specify title here. Please see this example for how to specify titles: [_data/guides.yml](https://github.com/kubernetes/kubernetes.github.io/blob/master/_data/guides.yml#L8) and [docs/whatisk8s.md](https://github.com/kubernetes/kubernetes.github.io/blame/master/docs/whatisk8s.md#L5)

Basically remove `title: Kompose` here and insert title in docs/tools/kompose/index.md
